common/scala/src/main/scala/whisk/common/TransactionId.scala

```diff
@@ -174,9 +174,12 @@ object TransactionId {
  * A thread-safe transaction counter.
  */
 trait TransactionCounter {
+    val numberOfInstances: Int
+    val instance: Int
+
     def transid(): TransactionId = {
-        TransactionId(cnt.incrementAndGet())
+        TransactionId(cnt.addAndGet(numberOfInstances))
     }
 
-    private val cnt = new AtomicInteger(1)
+    private val cnt = new AtomicInteger(numberOfInstances + instance)
```

Comment:
For example if you have N controllers:
The tid of invoker  0 will start with N, the tid of invoker 1 with N+1 and so on...
After the change in line 181, the tid is not incremented by one, but by N.
So the tid of controller0 will always be x*N, the tid of controller 1 will be x*N+1 and the tid of controller(N-1) will be x*N+(N-1).

So if you have two controllers, controller 0 will have all eval tids and invoker 1 all odd tids. So the tid is unique across controllers.